.github/actions/add-checks/action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'Add Status Checks'
+description: 'Add status check on a specified head-sha'
+inputs:
+  name:
+    description: 'Name of status check'
+    required: true
+  status:
+    description: >-
+      Status of check with possible values of 'queued', 'in_progress', 'completed'
+    required: true
+  head-sha:
+    description: 'The commit ID to add the checks to'
+    required: true
+  conclusion:
+    description: >-
+      Conclusion of status check with possible values of 'action_required',
+      'cancelled', 'failure', 'neutral', 'success', 'skipped', 'stale', 'timed_out'.
+      This is only required if status is 'completed'.
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/github-script@v6
+      env:
+        name: ${{ inputs.name }}
+        head_sha: ${{ inputs.head-sha }}
+        status: ${{ inputs.status }}
+        conclusion: ${{ inputs.conclusion }}
+      with:
+        script: |
+          const { name, head_sha, status, conclusion } = process.env;
+          const { owner, repo } = context.repo;
```

Comment:
Prefer process github context values through env vars, even if seemingly within our trust boundary, it's more secure.

Reply:
So technically, this is not prone to typical [script injection](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions) attacks, because these contexts do *not* take part in script generation process. `context` is a variable that is provided outside the script generation.

That said, I'm all in for having this in environment variable for the sake of consistency, and this is still better in case `context` variable is malformed (which is a different risk from script injection, but valid). We'll update to use env var here!